plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -422,18 +436,29 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
          std::mutex              mtx;
          std::deque<ro_trx_t>    queue;
       };
+      struct all_threads_exec_time_t {
+         std::mutex        mtx;
+         fc::microseconds  total_time{ 0 };
+
+         void add_time(const fc::microseconds& t) {
+            std::lock_guard<std::mutex>  g(mtx);
+            total_time += t;
+         }
+      };
```

Comment:
Seems like `std::atomic<int64_t> total_time_us{0}` would be much simpler and more efficient.

Reply:
My first version was `std::atomic<uint64_t>`, but I dropped it for concern of using correct microsecond type. But this is only for debug logging, I will go for simplicity.